File: Data Structures & Algorithms/max-water-container/submission-1.java
class Solution {
    public int maxArea(int[] heights) {
       int max = 0, left = 0, right = heights.length-1;
       while (left<right){
         int min_width = Math.min(heights[left], heights[right]);
         max = Math.max(min_width*(right-left), max);
         if(heights[left]>heights[right]){
            right--;
         }else{
            left++;
         }
       }
       return max;

    }
}

File: Data Structures & Algorithms/binary-tree-diameter/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    maxEdge = 0
    def diameterOfBinaryTree(self, root: Optional[TreeNode]) -> int:
      self.findDiameter(root)
      return self.maxEdge
        
    def findDiameter(self, root:Optional[TreeNode])-> int:
        if root is None:
            return 0
        
        left = self.findDiameter(root.left);
        right = self.findDiameter(root.right);

        current = max(left, right)+1

        self.maxEdge = max(left+right, self.maxEdge)

        return current